py/desispec/scripts/stdstars.py

```diff
@@ -637,26 +669,32 @@ def main(args) :
         scalefac=10**((model_magr - cur_refmag)/2.5)
 
         log.info('scaling {} mag {:.3f} to {:.3f} using scale {}'.format(ref_mag_name, model_magr, cur_refmag, scalefac))
-        normflux.append(model*scalefac)
+        normflux[star] = model*scalefac
 
-    # Now write the normalized flux for all best models to a file
-    normflux=np.array(normflux)
-
-    fitted_stars = np.where(chi2dof != 0)[0]
-    if fitted_stars.size == 0 :
-        log.error("No star has been fit.")
-        sys.exit(12)
+    if comm is not None:
+        linear_coefficients = comm.reduce(linear_coefficients, op=MPI.SUM, root=0)
+        redshift = comm.reduce(redshift, op=MPI.SUM, root=0)
+        chi2dof = comm.reduce(chi2dof, op=MPI.SUM, root=0)
+        fitted_model_colors = comm.reduce(fitted_model_colors, op=MPI.SUM, root=0)
+        normflux = comm.reduce(normflux, op=MPI.SUM, root=0)
 
-    data={}
-    data['LOGG']=linear_coefficients[fitted_stars,:].dot(logg)
-    data['TEFF']= linear_coefficients[fitted_stars,:].dot(teff)
-    data['FEH']= linear_coefficients[fitted_stars,:].dot(feh)
-    data['CHI2DOF']=chi2dof[fitted_stars]
-    data['REDSHIFT']=redshift[fitted_stars]
-    data['COEFF']=linear_coefficients[fitted_stars,:]
-    data['DATA_%s'%color]=star_colors[color][fitted_stars]
-    data['MODEL_%s'%color]=fitted_model_colors[fitted_stars]
-    data['BLUE_SNR'] = snr['b'][fitted_stars]
-    data['RED_SNR']  = snr['r'][fitted_stars]
-    data['NIR_SNR']  = snr['z'][fitted_stars]
-    io.write_stdstar_models(args.outfile,normflux,stdwave,starfibers[fitted_stars],data)
+    # Now write the normalized flux for all best models to a file
+    if rank == 0:
+        fitted_stars = np.where(chi2dof != 0)[0]
+        if fitted_stars.size == 0 :
+            log.error("No star has been fit.")
+            sys.exit(12)
```

Comment:
Please update this logic so that if it needs to exit with an error, then that information is broadcast to all ranks and they all exit together.  As-is I think we'd end up with rank 0 exiting but the others hanging around.  Since there isn't any barrier after this that might be ok, but it is fragile.  In other code we've had problems with one rank exiting with an error but the other ranks not exiting until slurm kills the job for a timeout.